src/coreclr/gc/gc.cpp

```diff
@@ -13320,6 +13320,17 @@ void gc_heap::age_free_regions (const char* msg)
     // If we are doing an ephemeral GC as a precursor to a BGC, then we will age all of the region
     // kinds during the ephemeral GC and skip the call to age_free_regions during the BGC itself.
     bool age_all_region_kinds = (settings.condemned_generation == max_generation) || is_bgc_in_progress();
```

Comment:
should be 
`bool age_all_region_kinds = (settings.condemned_generation == max_generation);`

Reply:
fixed now.